core/src/main/scala/schema/package.scala

```diff
@@ -28,6 +28,11 @@ package object schema {
   case class SchemaDefinitionProviderError(msg: String, maybeCause: Option[Throwable] = None)
       extends SchemaError(msg, maybeCause)
 
+  object SchemaDefinitionProviderError {
+    def apply(e: Throwable): SchemaDefinitionProviderError =
+      SchemaDefinitionProviderError(e.getMessage, Some(e))
+  }
+
```

Comment:
Maybe you could add a type alias for `Either[SchemaDefinitionProviderError, ?]` here?